modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/NavigationCardTag.java

```diff
@@ -0,0 +1,199 @@
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.taglib.clay.servlet.taglib;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.taglib.util.TagResourceBundleUtil;
+
+import java.util.Set;
+
+import javax.servlet.jsp.JspException;
+import javax.servlet.jsp.JspWriter;
+
+/**
+ * @author Marko Cikos
+ */
+public class NavigationCardTag extends BaseCardTag {
+
+	@Override
+	public int doStartTag() throws JspException {
+		setAttributeNamespace(_ATTRIBUTE_NAMESPACE);
+
+		setContainerElement((getHref() == null) ? "div" : "a");
+
+		return super.doStartTag();
+	}
+
+	public String getDescription() {
+		return LanguageUtil.get(
+			TagResourceBundleUtil.getResourceBundle(pageContext), _description);
+	}
+
+	public String getTitle() {
+		return LanguageUtil.get(
+			TagResourceBundleUtil.getResourceBundle(pageContext), _title);
+	}
+
+	public Boolean isSmall() {
+		return _small;
+	}
+
+	public void setDescription(String description) {
+		_description = description;
+	}
+
+	@Override
+	public void setHref(String href) {
+		setDynamicAttribute(StringPool.BLANK, "href", href);
+
+		super.setHref(href);
+	}
+
+	public void setSmall(Boolean small) {
+		_small = small;
+	}
+
+	public void setTitle(String title) {
+		_title = title;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_description = null;
+		_small = false;
+		_title = null;
+	}
+
+	@Override
+	protected String processCssClasses(Set<String> cssClasses) {
+		cssClasses.add("card");
+		cssClasses.add("card-interactive");
+		cssClasses.add("card-interactive-primary");
+		cssClasses.add("card-type-template");
+
+		if (_small) {
+			cssClasses.add("template-card-horizontal");
+		}
+		else {
+			cssClasses.add("template-card");
+		}
+
+		return super.processCssClasses(cssClasses);
+	}
+
+	@Override
+	protected int processEndTag() throws Exception {
+		if (!_small) {
+			JspWriter jspWriter = pageContext.getOut();
+
+			jspWriter.write("</span></span>");
+
+			_writeBody(jspWriter);
+		}
+
+		return super.processEndTag();
+	}
+
+	@Override
+	protected int processStartTag() throws Exception {
+		super.processStartTag();
+
+		JspWriter jspWriter = pageContext.getOut();
+
+		if (!_small) {
+			jspWriter.write("<span class=\"aspect-ratio\"><span class=\"");
+			jspWriter.write("aspect-ratio-item ");
+			jspWriter.write("aspect-ratio-item-center-middle ");
+			jspWriter.write("aspect-ratio-item-flush\">");
+
+			return EVAL_BODY_INCLUDE;
```

Comment:
> I was in two minds about adding support for custom content in tag. In Clay, we have support for custom content, but Lexicon does nor imply it. Lexicon states that card should have 'just a minimalistic image or icon'. This could be interpreted that we want an opinionated tag with imageSrc and imageAlt attributes, and fixed markup. In the end, I settled on custom content for extra flexibility and Clay markup consistency.

Well, with custom content being hard blocked, I guess I don't have to be in two minds anymore. I will need to add `imageSrc` and `imageAlt` to support images, per Lexicon guidelines.

Reply:
Yup, makes sense 😄